src/dec/bgi/bse_file_decoder.cc

```diff
@@ -0,0 +1,110 @@
+#include "dec/bgi/bse_file_decoder.h"
+#include "algo/range.h"
+#include "err.h"
+
+using namespace au;
+using namespace au::dec::bgi;
+
+static const bstr magic = "BSE 1.0\x00"_b;
+
+namespace
+{
+    class BseDecoder final
+    {
+    public:
+        BseDecoder(io::File &input_file);
+        bstr decode();
+
+        const u32 data_size = 64;
+    private:
+        s32 rand();
+        void check_sums() const;
+
+        bstr data;
+        s32 seed;
+        u8 check_sum;
+        u8 check_xor;
+    };
+}
+
+BseDecoder::BseDecoder(io::File &input_file)
+{
+    input_file.stream.skip(magic.size());
+    input_file.stream.skip(2);
+    check_sum = input_file.stream.read<u8>();
+    check_xor = input_file.stream.read<u8>();
+    seed = input_file.stream.read_le<s32>();
+    data = input_file.stream.read(data_size);
+}
+
+s32 BseDecoder::rand()
+{
+    const auto tmp = (((seed * 257 >> 8) + seed * 97) + 23) ^ -1496474763;
```

Comment:
Please encode -1496474763 as a positive number in hex, like `0xDEADBEEF` (I believe it's `0xA6CD9B75` in this particular case)